src/util/util_errors.h

```diff
@@ -190,6 +190,11 @@ enum sssd_errors {
 #define EOK ERR_OK
 #endif
 
+#if !defined(ETIME)
+// libc++ <errno>.h defines it this way
+#define ETIME 9935
```

Comment:
Why 9935 and not 62 which is used in `/usr/include/asm-generic/errno.h`?

Reply:
Does FreeBSD have `ETIMEDOUT`?

Reply:
Yes, it is, but SSSD wants to discern between `ETIME` and `ETIMEOUT` judging from https://github.com/SSSD/sssd/blob/ad7dc210f79e9b521ff26449d10a0348debff4a8/src/sss_client/idmap/sss_nss_idmap.h#L299-L300

From what I gather, SSSD uses `ETIME` as an errno when `poll(..., POLLOUT)` returns zero, that is timeouts. So to be on the safe side I did not `#define ETIME ETIMEOUT`, but made it a distinct value.

Reply:
It seems they are. I followed the following call chain:
```
sss_cli_send_req (uses ETIME)
sss_cli_make_request_nochecks
sss_nss_make_request_timeout
sss_get_ex
sss_nss_getpwnam_timeout
```
So, to get rid of ETIME we'd have to remove the ability to discern between timeout types from `sss_nss_idmap.h` consumers.

Reply:
`pthread_mutex_timedlock` using in 'idmap' returns ETIMEDOUT.
But I'm not sure why it is important to be able to distinguish those time timeouts in idmap API.
Strictly speaking "request timed out but was send to SSSD" is incorrect because "ETIME" can be returned already at "send" step.

Reply:
@sumit-bose, is it important to keep this difference? Or could we use 'ETIMEDOUT' everywhere?

Reply:
Maybe we could simply use:
```
#if !defined(ETIME)
#ifdef __FreeBSD__
#define ETIME 9935
#else
#define ETIME 62
#endif
#endif```
and leave this discussion, which is not actually related to FreeBSD, out of this PR?

Reply:
https://github.com/SSSD/sssd/pull/7954